dll/win32/imm32/imm.c

```diff
@@ -112,7 +112,7 @@ BOOL WINAPI ImmLoadLayout(HKL hKL, PIMEINFOEX pImeInfoEx)
 /***********************************************************************
  *		ImmFreeLayout (IMM32.@)
  */
-BOOL WINAPI ImmFreeLayout(DWORD dwUnknown)
+BOOL WINAPI ImmFreeLayout(DWORD_PTR dwUnknown)
```

Comment:
Looks like this dwUnknown isn't unknown anymore... It seems to be generally an `HKL hKL` to be freed (see line 174), where also two special values (that would be otherwise invalid otherwise) are reserved for specific behaviour, namely this `IFL_OFF` and `IFL_UNLOAD`.

Reply:
What would you name it?

Reply:
Named `dwHKLorIFL`.

Reply:
What about just using `HKL hKL` and documenting the fact that it also takes two special values?
This is not unheard of from other Win32 functions: for example,
https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowpos
the `hWndInsertAfter` parameter;
and as you can see it doesn't have a weird type/name like `DWORD_PTR dwHWNDorSpecialValue` etc.
so I think that you could do just the same in your case.
The special and undocumented `IFL_OFF` and `IFL_UNLOAD` values could as well be defined as:
```c
#define IFL_OFF     ((HKL)1)
#define IFL_UNLOAD  ((HKL)2)
```

Reply:
Renamed as `HKL_USE_NON_IME` and `HKL_FREE_ALL`.